### PR TITLE
Add TLS certs verification for Splunk HEC exporter

### PR DESCRIPTION
**Pull request by chaitanyaphalak**
**Description:** Adds support for TLS certificate verification while use splunk hec exporter

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3015

**Testing:** Tested on localhost

**Documentation:** Add documentation for new/replaced configuration options

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@chaitanyaphalak please sign the CLA

File: exporter/splunkhecexporter/README.md [hunk 1]
```diff
@@ -22,6 +22,14 @@ The following configuration options can also be configured:
 - `disable_compression` (default: false): Whether to disable gzip compression over HTTP.
 - `timeout` (default: 10s): HTTP timeout when sending data.
 - `insecure_skip_verify` (default: false): Whether to skip checking the certificate of the HEC endpoint when sending data over HTTPS.
+- `insecure` (default = false): When set to true disables verifying the server's certificate chain and host name. The
```
Comment (jrcamp) [inline exporter/splunkhecexporter/README.md:25]: I don't think insecure does what it says. I'll have to look into it separately. It's mainly for gRPC, for HTTP whether TLS is used is controlled by whether scheme is http/https. I think we can just delete documentation for insecure here.

**Review by jrcamp — CHANGES_REQUESTED** [review]
Thanks for doing this, was just looking into it recently. Would be great to use HTTPClientSettings eventually but that may require some changes to core. This shouldn't make moving to HTTPClientSettings any harder so I think it's a good improvement for now.

File: exporter/splunkhecexporter/README.md (hunk 1 above)
Comment (chaitanyaphalak) [reply exporter/splunkhecexporter/README.md:25]: yeah @jrcamp thats cool :)

**Review by jrcamp — APPROVED** [review]

**Review by jrcamp — CHANGES_REQUESTED** [review]
@chaitanyaphalak looks like tests still failing though

**Review by jrcamp — APPROVED** [review]

**Comment by chaitanyaphalak** [comment]
@jrcamp all the tests/checks have passed, we should merge this :) I cant merge it :)

**Pull request merged**